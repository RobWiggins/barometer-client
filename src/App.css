* {
  box-sizing: border-box;
  font-family: 'Open Sans', 'Source Sans Pro', 'Lato', sans-serif;
  color: #eee0c3;
}

.container {
  background-color: #0f1b30;
  color: #eee0c3;
}

html {
  max-width: 1000px;
  min-width: 340px;
  background-color: #0f1b30;

}

.app-header {
  padding-top: 15px;
  padding-bottom: 1px;
  padding-left: 5px;
  padding-right: 5px;
  background-color: #0f1b30;
}

.title-letter-holder {
  margin: 0;
  padding: 20px;
}

.app-title {
  font-weight: 500;
  font-family: 'Cormorant', 'Lora', 'Montserrat', 'Roboto', 'serif';
  font-size: 60px;
  margin-top: 5px;
}

/* rainbow title colors */
.blue {
  color: #405de6;
} 

.blue-purple {
  color: #5851db;
} 

.purple {
  color: #833ab4;
} 

.purple-pink {
  color: #c13584;
} 

.pink {
  color: #e1306c;
} 

.red {
  color: #fd1d1d;
} 

.red-orange {
  color: #f56040;
} 

.orange {
  color: #f77737;
} 

.orange-yellow {
  color: #fcaf45;
} 

.tagline {
  color: #eee0c3;
  font-weight: 500;
  font-family: 'Cormorant', 'Lora', 'Montserrat', 'Roboto', 'serif';
  font-size: 35px;
  margin-top: 5px;
  margin-bottom: 15px;
}

main {
  text-align: center;
  margin: auto;
}

#landingPane p {
  padding-left: 10px;
  padding-right: 10px;
}

.feature-summary {
  padding-top: 10px;
}

#search-home-img {
  width: 100%;
  height: auto; 
}

body {
  text-align: center;
  margin-left: 15px;
  margin-right: 15px;
  /* background-color: snow; */
  min-height: 100vh;
  border-radius: 1%;
  /* padding-bottom: 15px; */
  min-width: 290px;
  /* what to do here with body background */
}

#searchArea {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: space-between;
  padding-top: 20px;
}

.search img {
  height: 1em;
} 

.search button {
  border-radius: 0px 20px 20px 0px;
  height: 32px;
  cursor: pointer;
}

button:focus {
  outline-width: 0;
  outline-color: transparent;
}

#searchError,
#searchError > p {
  background-color: #ff9800;
  color: white;
  font-weight: 1.5em;
}

#searchError {
  margin-top: 20px;
  font-size: 18px;
}

.hidden {
  visibility: hidden;
}

.pastSearch {
  padding-left: 10px;
  padding-right: 10px;
  background-color: rgba(221, 221, 221, .5);
  border: none;
  color: black;
  text-align: center;
  font-size: 16px;
  margin: 4px 2px;
  transition: 0.3s;
  cursor: pointer;
  border-radius: 5%;
}

.pastSearch:hover {
  color: white;
  background-color: rgba(29, 161, 242, .6);
}

#searchHistory {
  padding-bottom: 15px;
}

#searchHistory h3 {
  margin-top: 20px;
}

#searchTweets {
  border-radius: 20px 0px 0px 20px;
  border: rgba(221, 221, 221, 1) solid 1px;
  padding: 2px;
  width: 220px;
  text-indent: 10px;
  margin: 0 auto;
  -webkit-tap-highlight-color: transparent;
  height: 32px; 
}

input:focus {
  outline-width: 0;
  outline-color: transparent;
}

.chart {
  min-width: 200px;
}

#emotionChart {
  padding: 10px;
}

#listTitle {
  padding-bottom: 10px;
  padding-top: 5px;
  font-size: 25px;
}

#tweets > ul {
  display: flex;
  justify-content: center;
}

#tweets li {
  font-family: 'Source Sans Pro', 'Open Sans', 'Lato', sans-serif;
  border-left: solid lightgray 1px;
  border-right: solid lightgray 1px;
  border-top: solid lightgray 1px;
  background-color: rgba(160, 204, 231, 0.03);
  border-radius: 1%;
  padding: 10px;
  margin-left: 8px;
  text-align: left;
  line-height: 1.4em;
}

li:last-child { 
  border-bottom: solid lightgray 1px; 
}

ul {
  list-style: none;
  padding-left: none;
  padding-inline-start: 0px;
}

@media only screen and (min-width: 768px) {

  main {
    text-align: center;
    margin: auto;
  }

  html {
    min-width: 290px;
    max-width: none;
  }

  #grid-holder {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 10px;
    grid-auto-rows: minmax(100px, auto);
    grid-auto-columns: 50%;
    grid-gap: 40px;
  }

  #emotionChart {
    padding: 10px;
    grid-column: 2;
    grid-row: 1;
  }

  #sentimentChart {
    padding-top: 15px;
    grid-column: 2;
    grid-row: 2;
  }

  #tweet-list {
    grid-column: 1;
    grid-row: 1 / 3;
    max-height: 800px;
    overflow: auto;
  }
}
