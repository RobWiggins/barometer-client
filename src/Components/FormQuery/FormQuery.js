import React from 'react';

class FormQuery extends React.Component {

  render() {
    return (
      <div id="searchArea">
        <form
          className="search"
          id="searchOne"
          onSubmit={e => {
            e.preventDefault();
            this.props.handleSubmitQuery(e.target.searchTweets.value);
          }}
        >
          <label htmlFor="searchTweets" id='search-label'>Search Twitter keywords</label>
          <input required maxLength="25" disabled={this.props.isSearchDisabled}
            id="searchTweets"
            type="text"
            name="searchTweets"
            placeholder="New York Yankees"
            onChange={e => this.props.handleSearch(e.target.value)}
          />
          <button disabled={this.props.isSearchDisabled}>Search</button>
          <p className={!this.props.isSearchDisabled ? "hidden" : ""} id="searchWarning">*query must be less than 25 characters</p>
        </form>
      </div>
    );
  }
}

export default FormQuery;
