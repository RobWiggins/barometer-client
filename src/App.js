import React from 'react';
// import { Route, Link } from 'react-router-dom';
import config from './config';
import './App.css';
import Header from './Components/Header/Header.js';
import FormQuery from './Components/FormQuery/FormQuery';
import EmotionChart from './Components/EmotionChart/EmotionChart';
import TweetList from './Components/TweetList/TweetList';
import ExampleChart from './Components/ExampleChart/ExampleChart';
import SentimentChart from './Components/SentimentChart/SentimentChart';
import ExampleSentimentChart from './Components/ExampleSentimentChart/ExampleSentimentChart';
import SearchError from './Components/SearchError/SearchError';
import SearchHistory from './Components/SearchHistory/SearchHistory'

class App extends React.Component {

  state = {
    watsonEmotionResults: null,
    tweets: null,
    queries: null,
    currentQuery: null,
    isSearchDisabled: false, // TODO not used at moment
    hasError: false,
  };

  /* retrieve past query history for all users */
  componentDidMount() {
    fetch(config.API_ENDPOINT + '/queries/history')
    .then(response => {
      if (!response.ok) {
        throw new Error({ message: 'error with getting history'});
      }
      return response.json()
    }).then(data => {
        console.log('QUERY HISTORY ON MOUNT: ', data.queries)
        this.setState( {
          queries: data.queries
        });
    })
    .catch(err => console.log(err.message))
  }
  
  /* Query valid and submitted, add query to history */
  /** TODO im doing this wrong, needs prev props */
  componentDidUpdate() {
    const body = JSON.stringify( {
      query: this.state.currentQuery
    });
    const options = { 
      method: 'POST', 
      headers: { 'content-type': 'application/json' },
      body,
    };

    /** TODO im doing this wrong */
    fetch(config.API_ENDPOINT + '/queries/history', options)
      .then(response => {
        if (!response.ok) {
          throw new Error({ message: 'error with retrieving history'});
        }
        return response.json()
      })
      .then(data => console.log('inserted query to history: ', data))
      .catch(err => console.log(err.message))
  }

  // returns false and disables search function if search > 25ch
  handleSearch(searchQuery) {
    // maximum search query length, allows limiting of watson targets
    //   const queryMaxLength = 25;
    //   if (searchQuery.length > queryMaxLength && !this.state.isSearchDisabled) {
    //     this.setState({
    //       isSearchDisabled: true,
    //   });
    //  } else if (searchQuery.length <= queryMaxLength && this.state.isSearchDisabled) {
    //     this.setState({
    //       isSearchDisabled: false,
    //     });
    //   }
  }

  /* TODO come back and wire up functionally with twitter retrieveTweets() */
  handleSubmitQuery = query => {

    console.log(query);
    /* try to communicate with backend */
    fetch(config.API_ENDPOINT + `/tweets/queries/${query}`) // how to send body?
      .then(response => {
        if (!response.ok) {
          throw new Error({ message: 'something seems to have gone wrong' });
        }
        return response.json();
      })
      .then(data => {
        // console.log('FROM THE FRONT END!!!!!', data);
        console.log('data.watsonEmotionResults: ', data.watsonEmotionResults);
        this.setState({
          watsonEmotionResults: data.watsonEmotionResults,
          tweets: data.duplicatesFiltered,
          hasError: false,
          currentQuery: data.currentQuery
        });
        
      })
      .catch(error =>
        this.setState({
          hasError: true,
        })
      ); // fix error message handling
  };

  render() {
    let isEmotionDataPresent = this.state.watsonEmotionResults ? true : false;
    let isTweetDataPresent = this.state.tweets ? true : false;
    console.log('isEmotionDataPresent: ', isEmotionDataPresent);

    let emotionChartDisplay, sentimentChartDisplay;
    if (isEmotionDataPresent) {
      emotionChartDisplay = (
        <EmotionChart watsonEmotionResults={this.state.watsonEmotionResults} />
      );
      sentimentChartDisplay = (
        <SentimentChart
          watsonEmotionResults={this.state.watsonEmotionResults}
        />
      );
    } else {
      emotionChartDisplay = <ExampleChart />;
      sentimentChartDisplay = <ExampleSentimentChart />;
    }

    let errorDisplay;
    if (this.state.hasError) {
      errorDisplay = <SearchError />;
    } else {
      errorDisplay = '';
    }

    return (
      <main className="main">
        <Header />
        <FormQuery
          isSearchDisabled={this.state.isSearchDisabled}
          handleSearch={this.handleSearch}
          handleSubmitQuery={this.handleSubmitQuery}
        />
        {errorDisplay}
        <SearchHistory queries={this.state.queries}></SearchHistory>
        <div id="grid-holder">
            {emotionChartDisplay}
            {sentimentChartDisplay}
            <TweetList tweets={this.state.tweets} />
        </div>
      </main>
    );
  }
}

export default App;
