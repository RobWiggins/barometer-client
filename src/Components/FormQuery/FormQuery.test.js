import React from 'react';
import ReactDOM from 'react-dom';
import FormQuery from './FormQuery';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../../App';

describe('testing FormQuery Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <FormQuery
        handleSearch={App.handleSearch}
        handleSubmitQuery={App.handleSubmitQuery}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a FormQuery component snapshot', () => {
    const wrapper = shallow(
      <FormQuery
        handleSearch={App.handleSearch}
        handleSubmitQuery={App.handleSubmitQuery}
      />
    ).find('#searchArea');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
