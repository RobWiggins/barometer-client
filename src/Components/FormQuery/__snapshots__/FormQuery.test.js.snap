// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing FormQuery Component renders a FormQuery component snapshot 1`] = `
<div
  id="searchArea"
>
  <form
    className="search"
    id="searchOne"
    onSubmit={[Function]}
  >
    <input
      aria-label="Keywords to search for tweets"
      aria-required="true"
      id="searchTweets"
      maxLength="25"
      name="searchTweets"
      placeholder="New York Yankees"
      required={true}
      type="text"
    />
    <button>
      <img
        alt="search"
        src="./static/search_icon.png"
      />
    </button>
  </form>
</div>
`;
